File: practice/fib.py
def fib(n,memo = {}):
    if n in memo:
        return memo[n]
    if n == 0 or n == 1:
        return 1
    else:
        memo[n]=fib(n-1,memo) + fib(n-2,memo)
        return memo[n]
print(fib(0))
print(fib(10))
print(fib(50))

File: practice/fib.cpp
# include<iostream>
using namespace std;

int fib(int n,array[n]){
    if (array[n] > 0) return array[n];
    if (n == 1 || n == 0) return n;
    else 
        array[n]=fib(n-1,array)+ fib(n-2,array);
        return array[n];
}

int main(){
    cout<< fib(10);
    return 0;
}